tests/unit/onyxTest.ts

```diff
@@ -1262,4 +1263,28 @@ describe('Onyx', () => {
                 connectionIDs.map((id) => Onyx.disconnect(id));
             });
     });
+
+    it('should upate Snapshot when its data changed', async () => {
```

Comment:
Let's add more tests for the functionality, and make tests more atomic. Some ideas:
- How would it work for keys that aren't collection?
- Test it separately for other methods, `merge`, `set`, `clear`

Reply:
@luacmartins do we allow snapshots to be not a collection?

Reply:
typo `update`